app/backend/src/couchers/servicers/admin.py

```diff
@@ -531,3 +532,17 @@ def ListUserIds(self, request, context, session):
             user_ids=user_ids[:page_size],
             next_page_token=str(user_ids[-1]) if len(user_ids) > page_size else None,
         )
+
+    def EditReferenceText(self, request, context, session):
+        reference = session.execute(select(Reference).where(Reference.id == request.reference_id)).scalar_one_or_none()
+
+        if reference is None:
+            context.abort(grpc.StatusCode.NOT_FOUND, errors.REFERENCE_NOT_FOUND)
+
+        if request.new_text.strip() == "":
```

Comment:
You can also do `if not request.new_text.strip()` in python (an empty string is falsey), but that's minor